WcaOnRails/app/models/abstract_cached_model.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+class AbstractCachedModel < ActiveRecord::Base
```

Comment:
> However I would not like to do code duplication just because the code is hard to read

I don't want to duplicate hard to read code. I want to _replace_ it with this easy to read code that's (I think) only possible in the concrete classes:

    def self.cfind(id)
      @@cache ||= {}
      @@cache[id] ||= self.find(id)
    end

This is also less fragile by handling any new records appearing in the DB.